evaluator/datasets/dataset.py

```diff
@@ -0,0 +1,163 @@
+import logging
+from typing import List
+
+from fastapi import APIRouter
+from fastapi import HTTPException, Response, status, Request
+from fastapi.param_functions import Body, Path
+
+from evaluator.datasets.models import Dataset, DatasetRequest
+from evaluator.evaluator.core.dataset_handler import DatasetHandler
+
+logger = logging.getLogger(__name__)
+router = APIRouter(prefix="/dataset")
+dataset_item_type = 'dataset'
+dataset_handler = DatasetHandler()
+
+
+@router.get(
+    "/dataset/{name}",
+    name="squad",
+    skill_type="extractive-qa",
+    metric="accuracy",
+    mapping={
+        "id_column": "id",
+        "question_column": "questions",
+        "context_column": "context",
+        "answer_column": "anwers"
+    },
+    response_datasetmodel={
+        200: {
+            "model": List[Dataset],
+            "description": "List of all datasets",
+        },
+        404: {"description": "The dataset could not be found"},
+    },
+    # response_datasetmodel=List[Dataset],
+)
+async def get_datasets():
+    """Returns a Datalist."""
+    datasets = [dataset.name for dataset in Dataset]
+
+    logger.debug("get_dataset {dataset_name}".format(datasets=datasets))
+    return datasets
+
+
+@router.put(
+    "/dataset/{name}",
+    name="squad",
+    skill_type="extractive-qa",
+    metric="accuracy",
+    mapping={
+
+        "id-column": "id",
+        "question-column": "questions",
+        "context-column": "context",
+        "answer-column": "answers"
+    },
+    responses={
+        200: {
+            "model": Dataset,
+            "name": "The dataset information",
+        },
+        400: {
+            "name": "Failed to create the dataset in the API database",
+        },
+        500: {
+            "name": "Failed to create the dataset in the backend.",
+        },
+    },
+    response_model=Dataset,
+)
+async def put_dataset(
+        request: Request,
+        dataset_name: str = Path(..., description="dataset name"),
+        fields: DatasetRequest = Body(..., description="The dataset fields"),
+        # get dataset
+        dataset_item_name=dataset_handler.get_dataset(dataset_name=get_datasets),
+        response: Response = None,
+
+):
+    # get existing dataset
+    schema = await dataset_item_name
+    success = False
+    if schema is None:
+        # it create a new  dataset
+        response.status_code = status.HTTP_201_CREATED
+
+    else:
+
+        response.status_code = status.HTTP_200_OK
+
+    if success:
+
+        return schema
+    else:
+        raise HTTPException(status_code=400)
+
+
+@router.delete(
```

Comment:
Auch hier eigentlich die gleichen Sachen wie in den anderen Funktionen. Skill-type, mapping etc. spielen für das Löschen ja garkeine Rolle.